src/braket/aws/aws_device.py

```diff
@@ -274,6 +275,79 @@ def arn(self) -> str:
         """str: Return the ARN of the device"""
         return self._arn
 
+    @property
+    def is_available(self) -> bool:
+        """bool: Return if the device is currently available"""
+        if self.status != "ONLINE":
+            return False
+
+        current_datetime_utc = datetime.utcnow()
+        for execution_window in self.properties.service.executionWindows:
+            weekday = current_datetime_utc.weekday()
+            current_time_utc = current_datetime_utc.time()
+
+            if (
+                execution_window.windowEndHour < execution_window.windowStartHour
+                and current_time_utc < execution_window.windowEndHour
+            ):
+                weekday = weekday - 1
+                if weekday == -1:
+                    weekday = 6
+
+            matched_day = execution_window.executionDay == ExecutionDay.EVERYDAY
+            matched_day = matched_day or execution_window.executionDay == ExecutionDay.EVERYDAY
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.WEEKDAYS and weekday < 5
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.WEEKENDS and weekday > 4
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.MONDAY and weekday == 0
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.TUESDAY and weekday == 1
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.WEDNESDAY and weekday == 2
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.THURSDAY and weekday == 3
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.FRIDAY and weekday == 4
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.SATURDAY and weekday == 5
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.SUNDAY and weekday == 6
+            )
```

Comment:
This suggestion made the code much more readable. I needed to add in a check to make sure that the execution day was in the ordered list, since EVERYDAY, WEEKDAYS, and WEEKENDS would have caused issues otherwise.